app/component/ItineraryPage.js

```diff
@@ -314,29 +244,66 @@ class ItineraryPage extends React.Component {
     fetchQuery(this.props.relayEnvironment, walkAndBikeQuery, planParams)
       .toPromise()
       .then(result => {
+        // filter plain walking / biking away
+        const bikeParkItineraries = transitItineraries(
+          result.bikeParkPlan?.itineraries,
+        );
+        const bikePublicItineraries = transitItineraries(
+          result.bikeAndPublicPlan?.itineraries,
+        );
+
+        let n1 = bikeParkItineraries.length;
+        let n2 = bikePublicItineraries.length;
+        if (n1 < 3) {
+          n2 = Math.min(6 - n1, n2);
+        } else if (n2 < 3) {
+          n1 = Math.min(6 - n2, n1);
+        } else {
+          n1 = 3;
+          n2 = 3;
+        }
```

Comment:
We could add a comment for what this is used for.

Reply:
done